feat: adds drag and drop component docs (#490)
zendeskgarden/website: src/pages/components/draggable.mdx

    </ul>
  </Misuse>
</Usage>

<!-- markdownlint-disable no-duplicate-header -->

## How to use it

### Default

The Draggable has a content area and a grip to communicate it can be dragged and dropped.

import DraggableDefault from '../../examples/draggable/DraggableDefault.tsx';
import DraggableDefaultCode from '!!raw-loader!../../examples/draggable/DraggableDefault.tsx';

<CodeExample code={DraggableDefaultCode}>
  <DraggableDefault />
</CodeExample>

### Bare

Borders can be removed using `isBare` prop.

import DraggableBare from '../../examples/draggable/DraggableBare.tsx';
import DraggableBareCode from '!!raw-loader!../../examples/draggable/DraggableBare.tsx';

<CodeExample code={DraggableBareCode}>
  <DraggableBare />
</CodeExample>

### Content

The Draggable content area can accept other components and text.

import DraggableContent from '../../examples/draggable/DraggableContent.tsx';
import DraggableContentCode from '!!raw-loader!../../examples/draggable/DraggableContent.tsx';

<CodeExample code={DraggableContentCode}>
  <DraggableContent />
</CodeExample>

### Disabled

The Draggable can be disabled to prevent interaction. It doesn't appear in the tab order,

> can't receive focus, and may not read aloud by a screenreader.

Reviewer: Just noticed, the wording is missing something.
```suggestion
can't receive focus, and may not be read aloud by a screenreader.
```

Author: Thank you my good sir.

import DropzoneDefault from '../../examples/draggable/DropzoneDefault.tsx';
import DropzoneDefaultCode from '!!raw-loader!../../examples/draggable/DropzoneDefault.tsx';

<CodeExample code={DropzoneDefaultCode}>
  <DropzoneDefault />
</CodeExample>

### Danger

Danger styling communicates that items dropped will be removed or deleted. This example uses [dnd-kit](https://github.com/clauderic/dnd-kit).

import DropzoneDanger from '../../examples/draggable/DropzoneDanger.tsx';
import DropzoneDangerCode from '!!raw-loader!../../examples/draggable/DropzoneDanger.tsx';

<CodeExample code={DropzoneDangerCode}>
  <DropzoneDanger />
</CodeExample>

## Configuration

<Configuration reactPackage={props.data.mdx.package} components={props.data.mdx.components} />

## API

### Draggable

<Component components={props.data.mdx.components} componentName="draggable" />

<PropSheet components={props.data.mdx.components} componentName="draggable" />

#### Draggable.Content

<Component components={props.data.mdx.subcomponents} componentName="draggable.content" />

#### Draggable.Grip

<Component components={props.data.mdx.subcomponents} componentName="draggable.grip" />

> Provides visual feedback that a `Draggable` can be grabbed and moved.

Reviewer 2: ```suggestion
Provides visual indication that a `Draggable` can be grabbed and moved.
```
I think of "feedback" as a response triggered by a user's action.

Author: ah fair point, agreed!

### DraggableList

<Component components={props.data.mdx.components} componentName="draggableList" />

<PropSheet components={props.data.mdx.components} componentName="draggableList" />

#### DraggableList.DropIndicator

<Component components={props.data.mdx.subcomponents} componentName="draggableList.dropIndicator" />

> Indicates where a `DraggableList.Item` can be inserted into a `DraggableList`.

Reviewer 2: ```suggestion
Indicates the position where a `DraggableList.Item` being dragged will be inserted into a `DraggableList`.
```

#### DraggableList.Item

<Component components={props.data.mdx.subcomponents} componentName="draggableList.item" />

Nest an Item within a `DraggableList` component.

### Dropzone

<Component components={props.data.mdx.components} componentName="dropzone" />

<PropSheet components={props.data.mdx.components} componentName="dropzone" />

#### Dropzone.Icon

<Component components={props.data.mdx.subcomponents} componentName="dropzone.icon" />

Nest a custom [icon](/design/icons) in `Dropzone.Icon`.

#### Dropzone.Message

<Component components={props.data.mdx.subcomponents} componentName="dropzone.message" />

> Nest content within `Dropzone.Message`. message will automatically center

Reviewer 2: ```suggestion
Nest content within a `Dropzone.Message` to apply appropriate styles.
```
(Less visual description to decouple our documentation from the visual design.)
